waltz-data/src/main/java/com/khartec/waltz/data/survey/SurveyRunDao.java

```diff
@@ -0,0 +1,132 @@
+package com.khartec.waltz.data.survey;
+
+import com.khartec.waltz.common.DateTimeUtilities;
+import com.khartec.waltz.common.StringUtilities;
+import com.khartec.waltz.model.EntityKind;
+import com.khartec.waltz.model.EntityReference;
+import com.khartec.waltz.model.HierarchyQueryScope;
+import com.khartec.waltz.model.IdSelectionOptions;
+import com.khartec.waltz.model.survey.*;
+import com.khartec.waltz.schema.tables.records.SurveyRunRecord;
+import org.jooq.DSLContext;
+import org.jooq.Record;
+import org.jooq.RecordMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.sql.Date;
+import java.util.Optional;
+
+import static com.khartec.waltz.common.Checks.checkNotNull;
+import static com.khartec.waltz.common.StringUtilities.join;
+import static com.khartec.waltz.common.StringUtilities.split;
+import static com.khartec.waltz.schema.Tables.SURVEY_RUN;
+
+@Repository
+public class SurveyRunDao {
+
+    private static final String ID_SEPARATOR = ",";
```

Comment:
v. minor:  semicolon slightly safer (comma occasionally used as a decimal point in some locales)